### PR TITLE
Firefox Accounts Integration

### LINKED ISSUE #292: Update first-run editor
When I first download the extension, I want to understand (or be reminded) of the purpose of Lockbox, in support of the first action: create the first entry.

**NOTE** This first entry is decoupled from creating an online account. It's a manual initial set-up, which is the end-result of the onboarding experience.

**NOTE** As explored in #209, securing a Lockbox is not a requirement for first run, in an attempt to allow users to get value prior to needing an account. This first run screen is the opportunity to explore what that value message needs to be to support creating a first entry.

# Acceptance criteria
1. It should have onboarding and support instructions to guide users through first time use
2. It should provide easy-to-find access to additional information
 
# Expected behavior
1. Upon download, a user sees the "0 entry screen" or first run experience as outlined in the visual design (below).
2. Creating a first entry is the primary call to action.
3. After the first entry is created, the user is guided to secure the account with a Firefox account.
4. There is persistent access to the "FAQ" via a navigation bar that links out to an external site with the FAQ details.

# Visual design
screenshot-2017-12-6 lockbox fx add-on preview mode - invision

# Context
In Alpha research, we discovered that the value and purpose of Lockbox needs to be more clearly communicated. This first run screen is where we can reinforce, and highlight, the message that prompted the Lockbox download.

# Tasks

*This relies upon FxA landing (#209) first.*
- [x] #209
- [X] Content strategy for first-run message
- [x] Updated Text
- [x] #331 
- [x] decision on "Already Using Lockbox? Sign in.."

### LINKED ISSUE #363: Replace master password with FxA
Placeholder for work happening at epic #209 and PR #362

### LINKED ISSUE #364: Upgrade from "guest mode"
Encrypting data with FxA as part of epic #209 and work at #362

### LINKED ISSUE #378: Add integration tests for FxA login, as well as doorhanger.
With the new door-hanger and fxa login flow landing, we will need to update the integration tests as well as expand their coverage. I am proposing a few new tests as well as a login flow update.

* [x] Update Login flow to accommodate FxA
  - [x] FxA Sign-in on welcome page.
  - [x] FxA Sign in on Home page from 'guest` mode.
  - [x] FxA Create account on Home page from 'guest' mode.
* [ ] Door-hanger
  - [ ] FxA Sign-in
  - [ ] Options button navigation
  - [x] Entry's matching
* [x] Guest Mode
  - [x] Add entry
  - [x] Delete entry

It would be nice to test the restart functionality, but I will get to that at the end if I have time as restarting Firefox isn't currently easy to do with selenium.

### PR DESCRIPTION
**Pull request by linuxwolf**
Still to be finished:

* [x] "sign in" from full-screen
* [x] "create account" from full-screen
* [x] "user info" on full-screen (additional features in #384)
* [x] open welcome on (fresh) install #292 
* [x] blow away database and open welcome on (upgrade) install
* [x] rewire metrics (1st pass)
* [x] cleanup
* updates from verification
  - [x] "linked account" copy on homepage
  - [x] remove "hamburger" from account summary
* [x] integration tests?

Resolves #363
Resolves #364
Resolves #292
Resolves #378

### CODE REVIEW

**Comment by linuxwolf** [comment]
What it looks like right now ...

Start in guest mode:
guest-mode

Start in FxA mode:
fxa-mode

New locked doorhanger:
doorhanger-signin

**Comment by sandysage** [comment]
So. awesome. loveit

**Comment by devinreams** [comment]
@linuxwolf This is awesome. 😍 

1. I updated the original description to tie this explicitly to #363 and #364 
2. I added this to the 0.1.4 milestone anticipating review and release next week
3. Given that, would you like an initial review from @jimporter yet?

File: src/webextension/background/datastore.js [hunk 1]
```diff
@@ -20,9 +20,11 @@ async function recordMetric(method, itemid, fields) {
   telemetry.recordEvent(method, "datastore", extra);
 }
 
-export default async function openDataStore() {
+export default async function openDataStore(cfg) {
   if (!datastore) {
+    cfg = cfg || {};
```
Comment (jimporter) [inline src/webextension/background/datastore.js]: Instead of this, I think we could just define the function as `openDataStore(cfg = {})`.

File: src/webextension/background/message-ports.js [hunk 2]
```diff
@@ -27,46 +27,85 @@ export default function initializeMessagePorts() {
...
+        // FIXME: be more implicit on saving account info
+        await accounts.saveAccount(browser.storage.local);
+        await updateBrowserAction({datastore});
+        if (!message.silent) {
+          await openView("manage");
+        }
+
+        return {};
+      });
+    case "upgrade":
```
Comment (jimporter) [inline src/webextension/background/message-ports.js]: `"upgrade"` doesn't seem like the right name for this, but I'm not sure what the best name would be. Maybe something like `"link_fxa"`?

File: src/webextension/list/manage/components/account-status.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import PropTypes from "prop-types";
+import React from "react";
+import { connect } from "react-redux";
```
Comment (jimporter) [inline src/webextension/list/manage/components/account-status.js]: Redux stuff should go in the `containers/` directory. For really simple stuff (this might count unless you plan on adding more functionality to it), you can put everything in `containers`, but for more-complex stuff, you should create a purely presentational components in `components/` and then do the Redux hookup via `connect()` in `containers/`.

File: src/webextension/list/manage/components/account-status.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import Button from "../../../widgets/button";
+
+export function AccountStatus({email, onClick, ...props}) {
```
Comment (jimporter) [inline src/webextension/list/manage/components/account-status.js]: I imagine this is just WIP, but you're not using `onClick` or `props`. You could probably remove the latter entirely (and add the former to the `Button`).

File: src/webextension/background/accounts/index.js [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+  }
+
+  async signOut() {
+    // TODO: implement a complete signout/forget
+    // TODO: something server side?
+    this.info = undefined;
+    return this;
+  }
+
+  status() {
```
Comment (jimporter) [inline src/webextension/background/accounts/index.js]: This name confused me a fair bit when reading code further down; I'd expect `status` to be an enum or something describing whether the user was logged in or not. Maybe something like `details`?

File: src/webextension/background/message-ports.js [hunk 6]
```diff
@@ -27,46 +27,85 @@ export default function initializeMessagePorts() {
 
   browser.runtime.onMessage.addListener(async (message, sender) => {
     switch (message.type) {
+    case "account_status":
+      return {account: getAccount().status()};
```
Comment (jimporter) [inline src/webextension/background/message-ports.js]: Maybe call this message type `"get_account_details"` or `"get_account"`? As above, "status" was confusing me, and adding "get" helps with consistency, since all the other message types are verbs.

File: src/webextension/list/actions.js [hunk 7]
```diff
@@ -33,6 +36,37 @@ export const HIDE_MODAL = Symbol("HIDE_MODAL");
 // other (i.e. FOO_STARTING and FOO_COMPLETED).
 let nextActionId = 0;
 
+export function accountStatus() {
```
Comment (jimporter) [inline src/webextension/list/actions.js]: As above, I'd call this `getAccountDetails` or `getAccount`; actions are verbs, and "status" was confusing to me.

File: src/webextension/list/manage/components/upgrade-account.css [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+.upgrade > menu {
+  margin: 0;
+  padding: 0;
+  display: flex;
+  flex-flow: row nowrap;
+  align-items: center;
+}
+.upgrade > menu > button {
```
Comment (jimporter) [inline src/webextension/list/manage/components/upgrade-account.css]: Nit: newline before this line.

File: src/webextension/list/manage/components/upgrade-account.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+ import { Localized } from "fluent-react";
```
Comment (jimporter) [inline src/webextension/list/manage/components/upgrade-account.js]: Nit: remove the extra whitespace at the start of this line.

File: src/webextension/list/manage/components/upgrade-account.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+import PropTypes from "prop-types";
+import React from "react";
+import { connect } from "react-redux";
```
Comment (jimporter) [inline src/webextension/list/manage/components/upgrade-account.js]: Similar to the `AccountStatus` component, this should be split into two files with a presentational component in `components/` and a Redux-connected component in `containers/`. I'd also probably put the implementations of `doCreate` and `doSignIn` in the Redux-ified version.

More generally, perhaps we should pipe our `sendMessage` stuff through Redux too? I'll have to think about that some more...

File: src/webextension/list/manage/components/upgrade-account.js [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    <section className={styles.upgrade}>
+      <Localized id="homepage-upgrade-title">
+        <h2>uPGRADe</h2>
+      </Localized>
+      <Localized id="homepage-upgrade-description">
+        <p>uPGRADe yOUr lOCKBOx</p>
+      </Localized>
+      <menu>
+        <Localized id="homepage-upgrade-action-create">
+          <Button id="homepage-upgrade-action-create"
```
Comment (jimporter) [inline src/webextension/list/manage/components/upgrade-account.js]: Avoid using IDs if you can help it. While I'm sure we could create sufficiently-unique IDs for single-use React components, I'd rather just not go there at all. If this is for tests and you need some help getting tests working without having IDs on elements, let me know and I can point you at some examples.

File: src/webextension/list/manage/reducers.js [hunk 12]
```diff
@@ -7,6 +7,18 @@ import { combineReducers } from "redux";
 import * as actions from "../actions";
 import { cacheReducer, listReducer } from "../reducers";
 
+export function accountReducer(state = { mode: "guest" }, action) {
```
Comment (jimporter) [inline src/webextension/list/manage/reducers.js]: Will we want this in the doorhanger as well? I notice below that the "account_status_updated" message is handled in `list/message-ports.js`, which applies to both the manager and the doorhanger, but the doorhanger won't have a reducer to respond to that action.

**Comment by pdehaan** [comment]
> _"You could probably use `const` instead of `let` in a bunch of places if you like. (I'm not sure I care one way or the other, but I suppose more const makes it harder to introduce certain kinds of bugs.)"_

ESLint has rules for this, if we want to enforce it:

- [**no-var**](https://eslint.org/docs/rules/no-var) &mdash; require `let` or `const` instead of var.
- [**prefer-const**](https://eslint.org/docs/rules/prefer-const) &mdash; If a variable is never reassigned, using the `const` declaration is better.

File: src/webextension/list/manage/components/account-status.js (hunk 4 above)
Comment (pdehaan) [reply src/webextension/list/manage/components/account-status.js]: This is weird. Not sure why ESLint wouldn't have detected the unused arguments here.

File: src/webextension/background/message-ports.js (hunk 2 above)
Comment (linuxwolf) [reply src/webextension/background/message-ports.js]: I went back and forth and ended up with "upgrade" due to my reading of the latest copy.  How about "bind_account"?

File: src/webextension/list/manage/components/account-status.js (hunk 3 above)
Comment (linuxwolf) [reply src/webextension/list/manage/components/account-status.js]: At some point in the near future, clicking the button should open a dropdown menu for "Sign out" and "Preferences".

I can move this whole thing to `containers/` for now, and address that refactor when we get to it.

File: src/webextension/list/manage/components/upgrade-account.js (hunk 11 above)
Comment (linuxwolf) [reply src/webextension/list/manage/components/upgrade-account.js]: This was for testing, so pointers would be much appreciated.

File: src/webextension/list/manage/reducers.js (hunk 12 above)
Comment (linuxwolf) [reply src/webextension/list/manage/reducers.js]: None of the visual prototypes have included this kind of information.

I can move the reducers and actions into `manage/`.

File: src/webextension/list/manage/components/upgrade-account.js (hunk 11 above)
Comment (jimporter) [reply src/webextension/list/manage/components/upgrade-account.js]: This is how I normally do it: https://github.com/mozilla-lockbox/lockbox-extension/blob/master/test/list/manage/components/item-details-test.js#L47-L48

Enzyme lets you find the `<Localized/>` element by its ID, and then you can just find the `<Button>` it wraps from there.

File: src/webextension/list/manage/reducers.js (hunk 12 above)
Comment (jimporter) [reply src/webextension/list/manage/reducers.js]: We can see what - if anything - Ryan has to say about this, but if it's not necessary, then the only thing we'd maybe want to do is have the message port listener in the doorhanger ignore the account update event. Even then, we could just have it fire and be a no-op...

Comment (linuxwolf) [reply src/webextension/list/manage/reducers.js]: word from @changecourse is that there are "no plans \[to include account information on the doorhanger]".

I'll shift things around so that `list/popup` doesn't have to see any of this.

**Comment by linuxwolf** [comment]
> *   You could probably use const instead of let in a bunch of places if you like. (I'm not sure I care one way or the other, but I suppose more const makes it harder to introduce certain kinds of bugs.)

`accounts/index` would definite benefit from more `const`.  It took a good amount of back-and-forth to get things working, and having `let` while doing that helped.  That logic is a lot more stable, so I can start changing those to `const`.

> *   Should we think about merging the unlock popup and the doorhanger? Maybe not in this PR, but if it's a medium-term goal, I can keep an eye out for stuff in this PR that would make it harder to make that change.

I think it would make more sense to merge unlock into the list doorhanger when we get closer to the "popup-less" signin with FxA.

File: src/webextension/list/manage/reducers.js (hunk 12 above)
Comment (linuxwolf) [reply src/webextension/list/manage/reducers.js]: Ok, so shifting all the account-details stuff "down" the tree requires a lot more changes than shifting the reducer up, and ignoring it.

I'm going to go with that, since it puts the actions and reducers at the same level, and is already ignored by things that don't care about it.

**Comment by linuxwolf** [comment]
What this looks like with the latest copy via @changecourse :

**welcome screen**
image

**guest mode**
image

**authenticated mode**
image

**unlock doorhanger**
image

**Comment by linuxwolf** [comment]
@jimporter all of the changes _should_ be addressing your review comments to date.

File: src/webextension/background/accounts/configs.json [hunk 13]
```diff
@@ -0,0 +1,32 @@
+{
+  "production": {
+    "client_id": "1b024772203a0849",
+    "redirect_uri": "https://2aa95473a5115d5f3deb36bb6875cf76f05e4c4d.extensions.allizom.org/",
+    "authorization_endpoint": "https://oauth.accounts.firefox.com/v1/authorization",
+    "token_endpoint": "https://oauth.accounts.firefox.com/v1/token",
+    "userinfo_endpoint": "https://profile.accounts.firefox.com/v1/profile",
```
Comment (rfk) [inline src/webextension/background/accounts/configs.json]: FWIW it should now be possible to automagically discover these URLs from https://accounts.firefox.com/.well-known/openid-configuration (or similar in dev) which has recently been updated with proper cross-origin support.

Comment (linuxwolf) [reply src/webextension/background/accounts/configs.json]: @rfk awesome!

Is it a showstopper if dynamic discover lands in the next iteration?  This work is really big, and I'm hoping discovery can wait until this settles ...

Comment (rfk) [reply src/webextension/background/accounts/configs.json]: No, not a showstopper at all, just making sure you're aware that CORS bug was fixed :-)

Comment (linuxwolf) [reply src/webextension/background/accounts/configs.json]: tracking this via #380

File: src/webextension/list/manage/reducers.js (hunk 12 above)
Comment (jimporter) [reply src/webextension/list/manage/reducers.js]: Will we want to be able to respond to FxA account events in the doorhanger? (e.g. if something caused us to be logged out, should we respond by locking the content of the doorhanger?) If so, we'd probably want to have the account reducer in the doorhanger.

**Comment by devinreams** [comment]
@m8ttyB As this looks nearly feature complete and @jimporter has his review comments in, is now an OK time to ask you to do an initial pass at functional testing this feature?

I know you all just went over it yesterday but want to get as much front-loaded as possible as early as possible this week.

**Comment by m8ttyB** [comment]
Thanks for the ping @devinreams, yes, I've begun testing this. \o/ @jrbenny35 has also made good progress w/ the integration tests.

**Comment by linuxwolf** [comment]
What the latest looks like:

welcome and guest mode:
welcome-to-guest

link to FxA:
link-fxa

unlock with FxA:
unlock

**Comment by devinreams** [comment]
@linuxwolf I just spoke with @irrationalagent. Is everything in a good spot for him to test the Telemetry events? It looks like it is but just wanted to confirm with you first.

**Comment by linuxwolf** [comment]
@irrationalagent @devinreams it should be good to go, yes.

**Comment by irrationalagent** [comment]
here's what I've been able to verify re:telemetry on my own machine:

1. clicking the fxa signin button from the welcome screen does **not** fire the `welcomeSignin` event but after auth w/ FxA I **did** get the `fxaUpgrade` event. 
screen shot 2017-12-07 at 12 15 13 pm

2. Clicking the fxa signin button from the manage screen **did** fire the `manageAcctSignin` click event, but after I successfully auth'd with FxA I **did  not** get the `fxaUpgrade` even though it did register visually in the manager **after I manually refreshed it**. The same was true for the create account click button and `manageAcctCreate` click event. Also, the `welcomeGuest` event **does** work.
screen shot 2017-12-07 at 12 19 37 pm

screen shot 2017-12-07 at 12 22 48 pm

Related to 2, I noticed that after I authorized with fxa, the manager did not refresh to show the upgrade - i had to close and re-open the manager to see it take effect. I also tried closing the manager before submitting my fxa credentials but the manager did not automatically re-open after authorization. 
After clicking the fxa links from the manager and authorizing is the manager supposed to refresh itself?  

Bottom line from me: Not having the auth events should **not** be a blocker for this. Maybe we can try to fix them at a later point. I will be able to tell if someone successfully authorized with fxa if their fxa ID is present, so in that sense we will have the information we need to analyze fxa usage even if we can't get the authorization events working properly before Austin.

**Comment by linuxwolf** [comment]
@irrationalagent 

That first one is a simple bug; will have fixed in next push.

That second one is not behavior I'm seeing on my working copy, but I wonder if the issue could be caused by outdated dependencies (especially an outdated `lockbox-datastore`).  Could you try running `npm install --no-copy` and see if that helps?

**Comment by irrationalagent** [comment]
`welcomeSignin` got fixed
screen shot 2017-12-07 at 1 19 17 pm

`fxaUpgrade` now fires after `manageAcctSignin`
screen shot 2017-12-07 at 1 22 00 pm

same with `manageAcctCreate` !
screen shot 2017-12-07 at 1 26 31 pm

thanks @linuxwolf !!!

File: src/webextension/locales/en-US/list.ftl [hunk 14]
```diff
@@ -69,6 +69,19 @@ homepage-greeting =
...
+    Creating a Firefox account - or adding { product-title } to an existing account - protects your logins
+    with the strongest encryption available.
+
+homepage-linkaccount-action-create = Create Account
+homepage-linkaccount-action-signin = { product-action-signin }
+
+homepage-accountlinked-title = Your logins are locked down tight!
+homepage-accountlinked-description =
+    { product-title } uses the strongest encryption available to protect
+    your logins - even for banking and other critical sites.
```
Comment (jimporter) [inline src/webextension/locales/en-US/list.ftl]: Ditto here re: em dashes verses hyphens.

File: src/webextension/background/accounts/index.js [hunk 15]
```diff
@@ -0,0 +1,250 @@
...
+    const error = new Error(`failed ${name} request: ${body.message || response.statusText}`);
+    error.errno = body.errno;
+    throw error;
+  }
+  return body;
+}
+
+export const GUEST = "guest";
+export const UNAUTHENTICATED = "unauthenticated";
+export const AUTHENTICATED = "authenticated";
```
[the remark this answers is not in the document]
Comment (linuxwolf) [reply src/webextension/background/accounts/index.js]: They need to be strings to cross the process boundary between front-end and back-end.

File: src/webextension/background/message-ports.js [hunk 16]
```diff
@@ -19,54 +19,128 @@ function broadcast(message, excludedSender) {
...
     case "initialize":
-      return openDataStore().then(async (ds) => {
-        await ds.initialize({
-          password: message.password,
-        });
-        await saveAuthorization(browser.storage.local);
-        await updateBrowserAction(ds);
+      return openDataStore().then(async (datastore) => {
+        await datastore.initialize();
+        // FIXME: be more implicit on saving account info
```
[the remark this answers is not in the document]
Comment (linuxwolf) [reply src/webextension/background/message-ports.js]: Good catch: I meant to change that for `XXX` or `TODO`.  For now it doesn't matter as much, but I suspect it'll matter a lot more when we implement cloud synchronization -- and tokens need refreshing.

File: src/webextension/list/actions.js [hunk 17]
```diff
@@ -33,6 +36,37 @@ export const HIDE_MODAL = Symbol("HIDE_MODAL");
...
+
+function getAccountDetailsCompleted(actionId, account) {
+  return {
+    type: GET_ACCOUNT_DETAILS_COMPLETED,
+    actionId,
+    account,
+  };
+}
+
+export function accountDetailsUpdated(account) {
```
[the remark this answers is not in the document]
Comment (linuxwolf) [reply src/webextension/list/actions.js]: I don't see how a distinction would help.  As far I can determine, the behavior and presentation needs to be the same in all cases.

File: src/webextension/list/reducers.js [hunk 18]
```diff
@@ -27,6 +27,18 @@ function maybeRemoveCurrentItem(state, action) {
   return {};
 }
 
+export function accountReducer(state = { mode: "guest" }, action) {
+  switch (action.type) {
+    case actions.GET_ACCOUNT_DETAILS_COMPLETED:
+      return {
+        ...state,
+        ...action.account,
+      };
```
[the remark this answers is not in the document]
Comment (linuxwolf) [reply src/webextension/list/reducers.js]: I'm getting old, and this helps my eyes and brain parse it.

File: src/webextension/background/accounts/index.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (jimporter) [reply src/webextension/background/accounts/index.js]: Oh well. :/

File: src/webextension/list/actions.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (jimporter) [reply src/webextension/list/actions.js]: Ok, just making sure.

File: test/background/accounts-test.js [hunk 19]
```diff
@@ -0,0 +1,307 @@
...
+    expect(getAccount.__GetDependency__("account"))
+          .to.equal(undefined);
+  });
+
+  describe("Account", () => {
+    let acct;
+    const unauthedInfo = {
+      uid: "1234",
+      access_token: "KhDtmS0a98vx6fe0HB0XhrtXEuYtB6nDF6aC-rwbufnYvQDgTnvxzZlFyHjB5fcF95AGi2TysUUyXBbprHIQ9g",
+      id_token: "eyJhbGciOiJSUzI1NiIsImtpZCI6IjBDWTBJb3RVVHBtWDFDbXJqMWNwcTB6aFBkd1NtalJSaWlJNzduMmMtMFkifQ.eyJ1aWQiOiIxMjM0IiwiaXNzIjoic2NvcGVkLWtleXMifQ.pX8I1LqCD849Gp0TzKcS6LM_fko0gc7wkSzBgPaxFDyF8AZrWn9-HTRoW-9YIuHujLzldbI1k34VeSHNM85vkPjm_AxbBKuXEiVJQdcCAxNjbSQmM1dOX6kZKwN4oDu8X4BB3CwQq5eXioYYiPur149O_I2bhFDuMBtQBoQosZtOScuKliXcURuWEwhYcnHe8axit0fQ0vd1FOJK3300hccqcZNoHGXrSVj42mdo_aSREOcwSUP4i0r0aCfJqnxai43uy1C5l54mSN1KzqGeasx60lWPU-Jm3gPm_2CXRWbfWxF3-OnxhMhSQiS90kefX81H03ZYVShDutsx55d0tQ",
```
[the remark this answers is not in the document]
Comment (linuxwolf) [reply test/background/accounts-test.js]: I originally meant for one to destructure into the other...

**Comment by linuxwolf** [comment]
@jimporter latest commits should address the concerns, minus a couple of things that might be overridden by Photon 2nd pass.

File: src/webextension/locales/en-US/list.ftl [hunk 20]
```diff
@@ -71,7 +71,7 @@ homepage-greeting =
 
 homepage-linkaccount-title = Add Serious Security & Convenience
 homepage-linkaccount-description =
-    Creating a Firefox account - or adding { product-title } to an existing account - protects your logins
+    Creating a Firefox account – or adding { product-title } to an existing account – protects your logins
```
Comment (jimporter) [inline src/webextension/locales/en-US/list.ftl]: One more micro-nit: I think em dashes usually don't have space around them, but feel free to double-check with Ryan.

Comment (linuxwolf) [reply src/webextension/locales/en-US/list.ftl]: The copy we got had spaces around them, and @changecourse thinks they should have spaces around them, too.

Comment (jimporter) [reply src/webextension/locales/en-US/list.ftl]: I looked it up and I guess I use Chicago style in my writing, whereas this is just AP style. :)

Comment (devinreams) [reply src/webextension/locales/en-US/list.ftl]: > this is just AP style

more-you-know

We use coast-based style guides around here, none of that mid-west stuff. 😏 🤠

**Comment by linuxwolf** [comment]
latest commit is merging master

**Comment by devinreams** [comment]
Thanks for the reviews @jimporter. Thanks for the integration test updates @jrbenny35. Thanks for all the testing @m8ttyB. And huge thanks to @linuxwolf for making this a reality.

This branch is as green as can be! ✅ :shipit:

**Comment by m8ttyB** [comment]
Bumping this to `pi-verified`, initial verification through our testing procedures (manual & automation) shows this as a go. We'll continue to test until it's go time.

**Comment by m8ttyB** [comment]
Telemetry - verified by @irrationalagent, we'll regression test again after all outstanding PRs have been merged
UX - verification ongoing by @changecourse and product
